File: quicksort.go
package main
import(
	"fmt"
	"math/rand"
)
func main(){
	var n int
	fmt.Printf("Enter the size of the array: ")
	fmt.Scanf("%d",&n)
	array:=make([]int,n)
	for i:=0;i<n;i++{
		fmt.Scanf("%d ",&array[i])
	}
	fmt.Println("\nUnsorted\n",array)
	quicksort(array)
	fmt.Println("Sorted\n",array)
}
func quicksort(array []int) []int{
	if len(array)<2{
		return array
	}
	left,right:=0,len(array)-1
	pivot:=rand.Int()%len(array)
	array[pivot],array[right]=array[right],array[pivot]
	for i:=range array{
		if array[i]<array[right]{
			array[left],array[i]=array[i],array[left]
			left++
		}
	}
	array[left],array[right]=array[right],array[left]
	quicksort(array[:left])
	quicksort(array[left+1:])
	return array
}

File: quickSort.js
function quickSort(arr, left, right){
   var len = arr.length, 
   pivot,
   partitionIndex;


  if(left < right){
    pivot = right;
    partitionIndex = partition(arr, pivot, left, right);
    
   //sort left and right
   quickSort(arr, left, partitionIndex - 1);
   quickSort(arr, partitionIndex + 1, right);
  }
  return arr;
}
//divide the array into parts
function partition(arr, pivot, left, right){
   var pivotValue = arr[pivot],
       partitionIndex = left;

   for(var i = left; i < right; i++){
    if(arr[i] < pivotValue){
      swap(arr, i, partitionIndex);
      partitionIndex++;
    }
  }
  swap(arr, right, partitionIndex);
  return partitionIndex;
}
//swap the elements
function swap(arr, i, j){
   var temp = arr[i];
   arr[i] = arr[j];
   arr[j] = temp;
}
//test    
let inputArray=[];
let size=prompt("enter size of array");
for(let i=0;i<size;i++)
  {
    inputArray[i]=prompt("enter sorted array");
    inputArray[i]=parseInt(inputArray[i]);
  }

console.log(quickSort(inputArray,0,size-1));